### PR TITLE
docs(stepfunctions): use `JsonPath.DISCARD` in place of null for input

### PR DESCRIPTION
**Pull request by paulhcsun**
Update docs for use of `sfn.JsonPath.DISCARD` in place of `null` in `TaskInput.fromObject`s field value for languages that do not support `null` like Python. 

Doc: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions.TaskInput.html#:~:text=to%20a%20task.-,static%20fromObject(obj),-public%20static%20fromObject

Closes https://github.com/aws/aws-cdk/issues/16253.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mrgrain** [comment]
Does `JsonPath.DISCARD` not work?

**Comment by paulhcsun** [comment]
@mrgrain `JsonPath.DISCARD` does work, I'm just not sure if the name of it is accurate for the purpose of replacing a `null` though, especially because the effect of using a `null` is not consistent across different cases. Taking one example provided in the issue:
```
"ItemResult": {
	"LastEvaluatedKey": JsonPath.DISCARD
     }
```
The name `DISCARD` here isn't the clearest of what we're trying to do but using `NULL_VALUE` is. 

Actually reading the [doc](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions.JsonPath.html#:~:text=Description-,static%20DISCARD,-string) on it, it could make sense (`Special string value to discard state input, output or result.`). Perhaps I'm just thinking of `DISCARD` in a different context. Let me know what you think though.

File: packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts [hunk 1]
```diff
@@ -142,7 +149,9 @@ export function recurseObject(obj: object | undefined, handlers: FieldHandlers,
 
   const ret: any = {};
   for (const [key, value] of Object.entries(obj)) {
-    if (typeof value === 'string') {
+    if (value === NULL_SUBSTITUTE) {
+      Object.assign(ret, handlers.handleNull(key, null));
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts:153]: Does the following not work?
```
if (value === null || value === undefined) {
     Object.assign(ret, handlers.handleNull(key, null));
}
```

Comment (paulhcsun) [reply packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts:153]: I think users want to specifically be able to distinguish `null` from `undefined` in typescript, but at the same time the two are the same in other languages like Python (`None`) so a magic value of some sort will satisfy both cases.

**Comment by mrgrain** [comment]
> Actually reading the [doc](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions.JsonPath.html#:~:text=Description-,static%20DISCARD,-string) on it, it could make sense (`Special string value to discard state input, output or result.`). Perhaps I'm just thinking of `DISCARD` in a different context. Let me know what you think though.

I believe this has been named based on this feature and assuming that the only time someone actually wants `null` in the State Machine definition is for this reason. It's still the only reference to `null` that I can find in the Developer Guide, but that doesn't mean much.

https://docs.aws.amazon.com/step-functions/latest/dg/input-output-resultpath.html

If you look at the PRs and issue involved in `JsonPath.DISCARD` you might get some more context on it. Look out for comments and PRs by @RomainMuller 

> @mrgrain `JsonPath.DISCARD` does work, I'm just not sure if the name of it is accurate for the purpose of replacing a `null` though, especially because the effect of using a `null` is not consistent across different cases. Taking one example provided in the issue:

If we find that the naming/documentation for `JsonPath.DISCARD` isn't working for our users, we can definitely think about improving this. Aside from sprucing up the docs, I could imagine adding an alias for `JsonPath.DISCARD`, e.g. `JsonPath.NULL`.

File: packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts (hunk 1 above)
Comment (mrgrain) [reply packages/aws-cdk-lib/aws-stepfunctions/lib/private/json-path.ts:153]: @paulhcsun's right. This isn't an issue in TS (at least not as far as I know), but in jsii (languages) which cannot distinguish between `null` and `undefined`. But ASL (as a JSON-ish schema) sometimes requires users to use `null`

**Review by kaizencc — APPROVED** [review]

**Pull request merged**